File: pm/src/main/java/org/sqlbuilder/pm/objects/PbRoleAlias.java
package org.sqlbuilder.pm.objects;

import org.sqlbuilder.common.Resolvable;
import org.sqlbuilder.common.Utils;
import org.sqlbuilder2.ser.Pair;

public class PbRoleAlias implements Resolvable<Pair<String, String>, Pair<Integer, Integer>>
{
	public String roleset;

	public String arg;

	@Override
	public String dataRow()
	{
		return String.format("%s,%s", Utils.nullableQuotedEscapedString(roleset), Utils.nullableQuotedEscapedString(arg));
	}

	@Override
	public Pair<String, String> resolving()
	{
		return new Pair<>(roleset, arg);
	}
}

File: pm/src/main/java/org/sqlbuilder/pm/objects/PbRoleAlias.kt
package org.sqlbuilder.pm.objects

import org.sqlbuilder.common.Resolvable
import org.sqlbuilder.common.Utils.nullableQuotedEscapedString
import org.sqlbuilder.pm.PmPbRoleResolvable
import org.sqlbuilder.pm.PmPbRoleResolved

class PbRoleAlias : Resolvable<PmPbRoleResolvable, PmPbRoleResolved> {

    var roleset: String? = null

    var arg: String? = null

    override fun dataRow(): String {
        return "${nullableQuotedEscapedString(roleset)},${nullableQuotedEscapedString(arg)}"
    }

    override fun resolving(): PmPbRoleResolvable {
        return Pair(roleset!!, arg!!)
    }
}
